### PR TITLE
Improve unit test for gnmi

### PR DESCRIPTION
**Pull request by ganglyu**
Signed-off-by: Gang Lv ganglv@microsoft.com

#### Why I did it
We should not run unit test without redis server.

#### How I did it
Check redis.sock, if there's no redis.sock, do not run unit test.

#### How to verify it
Run unit test for gnmi, and build image.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -77,6 +79,9 @@ check:
 	$(GO) get github.com/AlekSi/gocov-xml
 	gocov convert coverage-*.txt | gocov-xml -source $(shell pwd) > coverage.xml
 	rm -rf coverage-*.txt 
+else
+	$(info Please start redis server to run unit test)
```
Comment (qiluo-msft) [inline Makefile:83]: This will be treated as test passed? Is it expected?

File: Makefile [hunk 2]
```diff
@@ -24,6 +24,7 @@ endif
 GO_DEPS := vendor/.done
 PATCHES := $(wildcard patches/*.patch)
 PATCHES += $(shell find $(MGMT_COMMON_DIR)/patches -type f)
+REDIS_SOCK := $(shell if sudo [ -e /var/run/redis/redis.sock ]; then echo "exist"; else echo "noexist"; fi)
```
Comment (qiluo-msft) [inline Makefile:27]: Instead of checking a file, how about `redis-cli PING`?

File: Makefile (hunk 1 above)
Comment (ganglyu) [reply Makefile:83]: When we create a PR for sonic-gnmi repo, azure pipeline will start redis server, and we can run unit test.
But when we use sonic-buildimage to build, there's no redis server, and we should not run unit test.

File: Makefile [hunk 3]
```diff
@@ -65,7 +65,7 @@ else
 	$(GO) install -mod=vendor github.com/sonic-net/sonic-gnmi/gnoi_client
 endif
 
-check:
+ut:
```
Comment (qiluo-msft) [inline Makefile:68]: >ut [](http://example.com/codeflow?start=0&length=2)

`check_gotest` ?

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**